【BUG】Type errors

gen_swagger_api.ts:67:5 - error TS2353: Object literal may only specify known properties, and 'prettier' does not exist in type 'Partial<{ input: string; modular: boolean; output: string | false; url: string; spec: unknown; fileName: string; templates: string; templatePaths: { base: string; default: string; modular: string; original: string; custom: string; }; ... 64 more ...; update: (update: Partial<...>) => void; }>'.

67     prettier: prettierrc,
       ~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.2.9_magicast@0.3.5/node_modules/swagger-typescript-api/dist/lib.d.cts:1:25 - error TS2307: Cannot find module 'openapi-types' or its corresponding type declarations.

1 import { OpenAPI } from "openapi-types";
                          ~~~~~~~~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.2.9_magicast@0.3.5/node_modules/swagger-typescript-api/dist/lib.d.cts:4:8 - error TS1259: Module '"/Users/alaywn/github/alaywn/idis/modules/client/node_modules/.pnpm/@types+lodash@4.17.20/node_modules/@types/lodash/index"' can only be default-imported using the 'esModuleInterop' flag

4 import lodash from "lodash";
         ~~~~~~

  ../node_modules/.pnpm/@types+lodash@4.17.20/node_modules/@types/lodash/index.d.ts:14:1
    14 export = _;
       ~~~~~~~~~~~
    This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

../node_modules/.pnpm/swagger-typescript-api@13.2.9_magicast@0.3.5/node_modules/swagger-typescript-api/dist/lib.d.cts:697:5 - error TS2339: Property 'typeName' does not exist on type '{}'.

697     typeName,
        ~~~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.2.9_magicast@0.3.5/node_modules/swagger-typescript-api/dist/lib.d.cts:698:5 - error TS2339: Property 'schema' does not exist on type '{}'.

698     schema,
        ~~~~~~

../node_modules/.pnpm/swagger-typescript-api@13.2.9_magicast@0.3.5/node_modules/swagger-typescript-api/dist/lib.d.cts:699:5 - error TS2339: Property 'schemaPath' does not exist on type '{}'.

699     schemaPath
        ~~~~~~~~~~


Found 6 errors in 2 files.

--- a/.changeset/lazy-rules-guess.md
+++ b/.changeset/lazy-rules-guess.md
@@ -1,0 +1,10 @@
+---
+"swagger-typescript-api": patch
+---
+
+Move `@types/lodash` and `openapi-types` to dependencies
+
+These type packages are referenced by the published declarations, so consumers
+require them at install time for correct type resolution. Moving them from
+`devDependencies` prevents downstream TypeScript errors whilst having no
+runtime impact.

--- a/package.json
+++ b/package.json
@@ -49,6 +49,7 @@
   "dependencies": {
     "@biomejs/js-api": "3.0.0",
     "@biomejs/wasm-nodejs": "2.2.4",
+    "@types/lodash": "^4.17.20",
     "@types/swagger-schema-official": "^2.0.25",
     "c12": "^3.3.0",
     "citty": "^0.1.6",
@@ -57,6 +58,7 @@
     "js-yaml": "^4.1.0",
     "lodash": "^4.17.21",
     "nanoid": "^5.1.5",
+    "openapi-types": "^12.1.3",
     "swagger-schema-official": "2.0.0-bab6bed",
     "swagger2openapi": "^7.0.8",
     "typescript": "~5.9.2"
@@ -68,11 +70,9 @@
     "@tsconfig/node20": "20.1.6",
     "@tsconfig/strictest": "2.0.5",
     "@types/js-yaml": "4.0.9",
-    "@types/lodash": "4.17.20",
     "@types/node": "24.5.2",
     "@types/swagger2openapi": "7.0.4",
     "axios": "1.12.2",
-    "openapi-types": "12.1.3",
     "tsdown": "0.15.2",
     "typedoc": "0.28.13",
     "vitest": "3.2.4"

